File: src/core/jupiter/core/domain/concept/journals/journal_source.py
"""The source of a journal entry."""

from jupiter.core.framework.value import EnumValue, enum_value


@enum_value
class JournalSource(EnumValue):
    """The source of a journal entry."""

    USER = "user"
    GENERATED = "generated"

    @property
    def allow_user_changes(self) -> bool:
        """Whether the user can change the journal."""
        # Keep synced with ts:journal-source.ts
        return self == JournalSource.USER

File: src/webui/app/logic/domain/journal-source.ts
import { JournalSource } from "@jupiter/webapi-client";

export function journalSourceName(source: JournalSource) {
  switch (source) {
    case JournalSource.USER:
      return "User";
    case JournalSource.GENERATED:
      return "Generated";
  }
}

export function allowUserChanges(source: JournalSource) {
  // Keep synced with python:journal-source.py
  return source === JournalSource.USER;
}
